lib/HttpSignatureStrategy.js

```diff
@@ -211,13 +216,46 @@ function getDidPublicKey(publicKey, callback) {
   }, callback);
 }
 
+// 1. resolve key ID => should contain public key info including key material
+// (and owner)
+// 2. resolve owner ID and get identity info and list of public keys
+// 3. make sure key ID is listed in list of keys -- if so, verified
 function getHttpsPublicKey(publicKey, callback) {
-  // TODO: implement
-  callback(new BedrockError(
-    'URL scheme "https" is not supported.',
-    'HttpSignature.UnsupportedUrlScheme', {
-      'public': true,
-      httpStatusCode: 400,
-      scheme: 'https'
-    }));
+  async.auto({
+    dereferencePublicKey: function(callback) {
+      request.get(publicKey, function(err, res, body) {
+        callback(err, res.body);
+      });
+    },
+    dereferenceOwner: ['dereferencePublicKey', function(callback, results) {
+      var dereferencedKey = results.dereferencePublicKey;
+      var owner = dereferencedKey.owner;
+      request.get(owner, function(err, res, body) {
```

Comment:
Nevermind, `request.defaults` gives you a new instance of request, so that's ok. We just need to be more specific about the MIME type we want --- it isn't just JSON.